### PR TITLE
Makefile

### PR DESCRIPTION
**Pull request by jdmoreira**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+CC:=cc
```
Comment (jcfs) [inline Makefile:1]: Was the usage of clang considered in the lastest tech meeting?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+CFLAGS:= -Wall -pedantic -std=c99
```
Comment (jcfs) [inline Makefile:3]: c99? Really? plz...

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+FRAMEWORKS:=
```
Comment (jcfs) [inline Makefile:5]: No frameworks for now, lets keep the file simple.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+LIBRARIES:=
```
Comment (jcfs) [inline Makefile:6]: Same as above.

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+LDFLAGS=$(LIBRARIES) $(FRAMEWORKS)
```
Comment (jcfs) [inline Makefile:8]: If there are not LIBS nor FRAMEWORK we don't need LDFLAGS, remove it.

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+BIN:= main
```
Comment (jcfs) [inline Makefile:10]: Shouldn't the binary file name be discussed first?

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+SRCS:= $(BIN:=.c) $(wildcard src/*.c)
```
Comment (jcfs) [inline Makefile:11]: Single source. Keep it simple.

File: Makefile [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+OBJS:= $(SRCS:=.o)
```
Comment (jcfs) [inline Makefile:12]: This one is ok.

File: Makefile [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+DEPS:= $(wildcard include/*.h)
```
Comment (jcfs) [inline Makefile:13]: No deps so far.

File: Makefile [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+$(BIN): $(OBJS) $(DEPS)
```
Comment (jcfs) [inline Makefile:15]: Target seems good, but shouldn't have deps.

File: Makefile [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+	$(CC) $(OBJS) $(LDFLAGS) -o $(@)
```
Comment (jcfs) [inline Makefile:16]: See above.

File: Makefile [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+%.c.o: %.c
```
Comment (jcfs) [inline Makefile:18]: Seems fine.

File: Makefile [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+	$(CC) $(CFLAGS) -c $< -o $@
+
+.PHONY: clean
```
Comment (jcfs) [inline Makefile:21]: 👍

**Review by joaoappereira — APPROVED** [review]

File: Makefile [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+clean:
+	rm -f *.o $(wildcard src/*.o)
```
Comment (jcfs) [inline Makefile:23]: Nothing to comment here.

**Review by jcfs — CHANGES_REQUESTED** [review]
Hmm... seems that we are overengineering a little bit, no?

**Review by jcfs — APPROVED** [review]

**Comment by jcfs** [comment]
Approved after IRL discussion.

**Comment by jdmoreira** [comment]
Thanks for the feedback 🙏
Great culture we are building at megacorp

**Pull request merged**